agenda-services/src/main/java/org/exoplatform/agenda/plugin/AgendaGuestUserIdentityProvider.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2022 eXo Platform SAS.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.exoplatform.agenda.plugin;
+
+import javax.mail.internet.InternetAddress;
+
+import org.apache.commons.lang.StringUtils;
+
+import org.exoplatform.social.core.identity.IdentityProvider;
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.identity.model.Profile;
+
+public class AgendaGuestUserIdentityProvider extends IdentityProvider<InternetAddress> {
+
+  public static final String NAME = "AGENDA_GUEST_USER";
```

Comment:
```suggestion
  public static final String NAME = "GUEST_USER";
```
IMO, we can use generic name to allow using this IdentityProvider outside Agenda when other addons will need to interact with guest users.
For now, we can keep this IdentityProvider in agenda, but when another addon will need this identity provider, we will move it in social, by example. WDYT ?

Reply:
Yes, it is ok for me.
Done